pyTigerGraph/pyTigerGraph.py

```diff
@@ -960,54 +987,151 @@ def runInterpretedQuery(self, queryText, params=None):
+        See example in `edgeSetToDataFrame`.
+        
+        A vertex set has this structure:
+        [
+            {
+                "v_id": <vertex_id>,
+                "v_type": <vertex_type_name>,
+                "attributes":
+                    {
+                        "attr1": <value1>,
+                        "attr2": <value2>,
+                         ⋮
+                    }
+            },
+                ⋮
+        ]
+        
+        See: https://docs.tigergraph.com/dev/gsql-ref/querying/declaration-and-assignment-statements#vertex-set-variable-declaration-and-assignment
         """
-        data = self.getVerticesById(vertexType, vertexIds)
-        df = pd.DataFrame(data)
-        df = pd.concat([df["v_id"], pd.DataFrame(df["attributes"].tolist())], axis=1)
-        return df
-
-    def upsertVertexDataframe(self, df, vertexType, v_id=None, attributes=None):
-        """Upserts vertices from a Pandas data frame.
+        df = pd.DataFrame(vertexSet)
+        cols = []
+        if withId:
+            cols.append(df["v_id"])
+        if withType:
+            cols.append(df["v_type"])
+        cols.append(pd.DataFrame(df["attributes"].tolist()))
+        return pd.concat(cols, axis=1)
+
+    def edgeSetToDataFrame(self, edgeSet, withId=True, withType=False):
+        """Converts an edge set to Pandas DataFrame
+
+        Edge sets contain instances of the same edge type. Edge sets are not generated "naturally" like vertex sets, you need to collect edges in (global) accumulators,
+            e.g. in case you want to visualise them in GraphStudio or by other tools.
+        Example:
+        
+            SetAccum<EDGE> @@edges;
+            start = {Country.*};
+            result =
+                SELECT t
+                FROM   start:s -(PROVINCE_IN_COUNTRY:e)- Province:t
+                ACCUM  @@edges += e;
+            PRINT start, result, @@edges;
+
+        The `@@edges` is an edge set.
+        It contains for each edge instance the source and target vertex type and ID, the edge type, an directedness indicator and the (optional) attributes.
+        Note: `start` and `result` are vertex sets.
+
+        An edge set has this structure:
+        [
+            {
+                "e_type": <edge_type_name>,
+                "from_type": <source_vertex_type_name>,
+                "from_id": <source_vertex_id>,
+                "to_type": <target_vertex_type_name>,
+                "to_id": <targe_vertex_id>,
+                "directed": <true_or_false>,
+                "attributes":
+                    {
+                        "attr1": <value1>,
+                        "attr2": <value2>,
+                         ⋮
+                    }
+            },
+                ⋮
+        ]
+        """
+        df = pd.DataFrame(edgeSet)
+        cols = []
+        if type(withId) == bool:
```

Comment:
Why not `if withId`? Wouldn't this add them in even on `withId == False`?

Reply:
Fixed.
Originally I intended to allow users to select which ones of the `from_type`, `from_id`, `to_type`, `to_id` they wanted to include, so I wanted to check if the `withID` is a string or list with one of these key names; or bool to indicate adding all or none. But then I decided that it would be too much complexity for the users and it's easier for them simply ignore what they do not need. But the code hasn't been changes back to `if withID:`. Until now.